cardinal-components-base/src/main/java/nerdhub/cardinal/components/api/component/extension/CopyableComponent.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Cardinal-Components-API
+ * Copyright (C) 2019-2020 GlassPane
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package nerdhub.cardinal.components.api.component.extension;
+
+import nerdhub.cardinal.components.api.ComponentType;
+import nerdhub.cardinal.components.api.component.Component;
+import net.minecraft.nbt.CompoundTag;
+
+/**
+ * A component that can copy its data from another component of the same type.
+ *
+ * @param <C> this component's type
+ * @since 2.3.0
+ */
+public interface CopyableComponent<C extends Component> extends TypeAwareComponent {
+    default void copyFrom(C other) {
+        this.fromTag(other.toTag(new CompoundTag()));
```

Comment:
The default impl would work always, but you probably also want to override it because the `C` -> `NBT` -> `C` conversion is slower than a simple `C` -> `C` copy that would be implemented by whoever implements the interface. Seems like reasonable behavior, but you might want to document that.

Reply:
Good point.